astrosat/utils/utils_logging.py

```diff
@@ -62,3 +65,52 @@ def emit(self, record):
                 trace=trace,
             )
             db_record.tags.add(*tags)
+
+
+def format_elasticsearch_timestamp(time):
+    "Renders a timestamp in the format expected by elasticsearch"
+    return LogstashFormatterBase.format_timestamp(time)
+
+
+class ElasticsearchDocumentLogFormatter(LogstashFormatterBase):
+    """
+    Formats a json log record with additional fields needed for elasticsearch.
+    Input records MUST be JSON.
+    Based on LogstashFormatterBase/Version1 from python-logstash
+    https://github.com/vklochan/python-logstash/blob/master/logstash/formatter.py
+    """
+
+    def __init__(self, constant_fields=dict()):
+        self.constant_fields = constant_fields
+
+    def format(self, record):
+
+        record_fields = json.loads(record.getMessage())
+
+        meta_fields = {
+            "@timestamp": format_elasticsearch_timestamp(record.created),
+            "@version": 1,
+        }
+
+        # Merge analytics, constant field overrides and meta fields into one dict
+        logstash_message = {**record_fields, **self.constant_fields, **meta_fields}
+
+        return self.serialize(logstash_message)
+
+
+class AstrosatAppTCPLogstashLogHandler(TCPLogstashHandler):
+    """
```

Comment:
Why is this called `AstrosatAppTCPLogstashLogHandler`?  Is it really specific to _Astrosat_, or is it specific to _ElasticSearch_ (b/c it uses the above formatter)?

Reply:
It's specific to Astrosat Apps, because it adds the constant fields expected by the `app-analytics` logstash instance for routing documents into indices: https://github.com/astrosat/orbis-infrastructure/blob/master/terraform/main/astrosat-elk/logstash.tf#L66-L68

do you know of a better way to do this?

Reply:
Your explanation makes sense.